#! /bin/sh /usr/share/dpatch/dpatch-run

## 22_vdr-playerepg.dpatch by Stefan Hülswitt <s.huelswitt@gmx.de>
## http://vdrportal.de/board/thread.php?postid=348701#post348701
##
## Thomas Günther <tom@toms-cafe.de>:
##   - adapted to vdr-1.3.31
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes mp3 plugin problems with live background and EPG scan enabled.

@DPATCH@
diff -u vdr-1.3.29-orig/device.c vdr-current-1.3.29/device.c
--- vdr-1.3.29-orig/device.c	2005-08-14 12:52:08.000000000 +0200
+++ vdr-current-1.3.29/device.c	2005-08-24 19:17:13.000000000 +0200
@@ -847,9 +849,9 @@
 {
 }
 
-bool cDevice::Replaying(void) const
+bool cDevice::Replaying(bool ignoreLivePlayer) const
 {
-  return player != NULL;
+  return player!=NULL && (!ignoreLivePlayer || player->playMode==pmAudioVideo || player->playMode==pmAudioOnlyBlack);
 }
 
 bool cDevice::AttachPlayer(cPlayer *Player)
diff -u vdr-1.3.29-orig/device.h vdr-current-1.3.29/device.h
--- vdr-1.3.29-orig/device.h	2005-08-13 13:44:13.000000000 +0200
+++ vdr-current-1.3.29/device.h	2005-08-24 18:58:32.000000000 +0200
@@ -460,7 +462,7 @@
        ///< to a complete packet with data from the next call to PlayPes().
        ///< That way any functions called from within PlayPes() will be
        ///< guaranteed to always receive complete PES packets.
-  bool Replaying(void) const;
+  bool Replaying(bool ignoreLivePlayer=false) const;
        ///< Returns true if we are currently replaying.
   void StopReplay(void);
        ///< Stops the current replay session (if any).
diff -u vdr-1.3.31-orig/eitscan.c vdr-current-1.3.31/eitscan.c
--- vdr-1.3.31-orig/eitscan.c	2005-08-28 18:46:14.000000000 +0200
+++ vdr-current-1.3.31/eitscan.c	2005-09-01 01:17:13.000000000 +0200
@@ -150,7 +150,7 @@
                          if (!Channel->Ca() || Channel->Ca() == Device->DeviceNumber() + 1 || Channel->Ca() >= 0x0100) {
                             if (Device->ProvidesTransponder(Channel)) {
                                if (!Device->Receiving()) {
-                                  bool IsPrimaryDeviceReplaying = Device == cDevice::PrimaryDevice() && Device->Replaying() && cTransferControl::ReceiverDevice() != cDevice::PrimaryDevice();
+                                  bool IsPrimaryDeviceReplaying = Device == cDevice::PrimaryDevice() && Device->Replaying(true) && cTransferControl::ReceiverDevice() != cDevice::PrimaryDevice();
                                   if (Device != cDevice::ActualDevice() || (Device->ProvidesTransponderExclusively(Channel) && (IsPrimaryDeviceReplaying || now - lastActivity > Setup.EPGScanTimeout * 3600))) {
                                      if (!IsPrimaryDeviceReplaying && Device == cDevice::ActualDevice() && !currentChannel) {
                                         if (cTransferControl::ReceiverDevice())
